tools/editor/project_manager.cpp

```diff
@@ -1103,16 +1103,48 @@ void ProjectManager::_install_project(const String& p_zip_path,const String& p_t
 }
 
 void ProjectManager::_files_dropped(StringArray p_files, int p_screen) {
-	bool import_project_file = false;
+	Set<String> folders;
+	DirAccess *da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
 	for (int i = 0; i < p_files.size(); i++) {
-		if (p_files[i].ends_with("engine.cfg")) {
-			npdialog->import_from_file(p_files[i]);
-			import_project_file = true;
+		String file = p_files[i];
+		folders.insert(da->dir_exists(file) ? file : file.get_base_dir());
+	}
+	memdelete(da);
+	multiple_folders.resize(0);
+	if (folders.size()>0) {
+		for (Set<String>::Element *E=folders.front();E;E=E->next()) {
+			multiple_folders.append(E->get());
+		}
+
+		bool confirm = true;
+		if (multiple_folders.size()==1) {
+			DirAccess *dir = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+			if (dir->change_dir(multiple_folders[0])==OK) {
+				dir->list_dir_begin();
+				String file = dir->get_next();
+				while(confirm && file!=String()) {
+					if (!da->current_is_dir() && file.ends_with("engine.cfg")) {
+						confirm = false;
+					}
+					file = dir->get_next();
+				}
+				dir->list_dir_end();
+			}
+			memdelete(dir);
+		}
+		if (confirm) {
+			multi_scan_ask->set_text(TTR("Are you sure to scan " + itos(multiple_folders.size()) + " folders?"));
+			multi_scan_ask->popup_centered_minsize();
+		} else {
+			_scan_multiple_folders();
 		}
 	}
-	if (!import_project_file && p_files.size() > 0) {
-		scan_dir->set_current_dir(p_files[0]);
-		scan_dir->popup_centered_ratio();
+}
+
+void ProjectManager::_scan_multiple_folders()
```

Comment:
Wouldn't it be cleaner/simpler to just pass `StringArray p_folders` as argument here and drop the member var `multiple_folders` altogether?